src/app/search/loadouts/search-filters/freeform.ts

```diff
@@ -41,6 +42,106 @@ function isLoadoutCompatibleWithStore(loadout: Loadout, store: DimStore | undefi
   return !store || isClassCompatible(loadout.classType, store.classType);
 }
 
+// Helper object to tell us useful things about all the DIM items we get back from a loadout
+// Primarily we are interested in items that contribute to the power level
+class ResolvedEquipment {
```

Comment:
We generally don't use classes in DIM code, for a number of reasons. Not that it's forbidden, but in general it tends not to be necessary.

Reply:
Alright - I don't use JS/TypeScript in my day job so I was just kind of winging it. Is there a preferred pattern that you do use? Or another place in DIM that I can use as reference?

Reply:
This turned out to have been a non-issue, after using `Object.groupBy`